calico_test/tests/st/utils/docker_host.py

```diff
@@ -28,14 +28,30 @@ class DockerHost(object):
     """
     A host container which will hold workload containers to be networked by
     Calico.
+
+    :param calico_node_autodetect_ip: When set to True, the test framework
+    will not perform IP detection, and will run `calicoctl node` without explicitly
+    passing in a value for --ip. This means calico-node will be forced to do its IP detection.
     """
     def __init__(self, name, start_calico=True, dind=True,
                  additional_docker_options="",
                  post_docker_commands=["docker load -i /code/calico-node.tar",
-                                       "docker load -i /code/busybox.tar"]):
+                                       "docker load -i /code/busybox.tar"],
+                 calico_node_autodetect_ip=False):
         self.name = name
         self.dind = dind
         self.workloads = set()
+        self.ip = None
+        """
+        An IP address value to pass to calicoctl as `--ip`. If left as None, no value will be passed,
+        forcing calicoctl to do auto-detection.
+        """
+
+        self.ip6 = None
+        """
+        An IPv6 address value to pass to calicoctl as `--ipv6`. If left as None, no value will be passed,
```

Comment:
Actually we only auto-detect IPv4 - so if this is left as none we don't configure the node to use IPv6.